### PR TITLE
Mikes branch

### PR DESCRIPTION
**Pull request by MikeMurpher**
@cryptod00c . If you could maybe have a look at the hook and the table css.

The table end column is hidden with no wallet connected through wagmi, when connected it will pull through the user rewards from that printer pool

image

### CODE REVIEW

File: components/printerPoolBalance/printerSection.tsx [hunk 1]
```diff
@@ -34,14 +36,40 @@ function ContractItem(props: ContractItemProps) {
...
       </header>
-      <ul
-        role="list"
-        className="mt-5 mb-5 grid grid-cols-1 gap-6 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-6"
-      >
-        {subItems.map((s: SubItemProps, i: number) => {
-          return <SubItem key={i} {...s} />;
-        })}
-      </ul>
+      <table className="divide-zinc-900/7.5 my-4 min-w-full divide-y rounded-lg bg-white">
```
Comment (MikeMurpher) [inline components/printerPoolBalance/printerSection.tsx:39]: @cryptod00c  start of table

File: components/printerPoolBalance/printerSection.tsx [hunk 2]
```diff
@@ -51,53 +79,67 @@ interface SubItemProps {
...
   } = props;
 
-  const { rewards } = usePrinterPoolData(contractAddress, chain, decimals);
+  const { rewardBalance, stakedBalance, hasUserLimit, pendingReward } =
+    usePrinterPoolData(contractAddress, chainId, decimals);
 
   return (
-    <li className="col-span-1 flex flex-col overflow-hidden rounded-lg bg-white text-center shadow">
-      <div className="flex flex-1 flex-col px-4 py-3">
+    <tr>
```
Comment (MikeMurpher) [inline components/printerPoolBalance/printerSection.tsx:105]: @cryptod00c start of rows

File: lib/hooks/usePrinterPoolData.tsx [hunk 3]
```diff
@@ -1,34 +1,70 @@
...
+        abi: PerpetualPrintingPool,
+        functionName: 'hasUserLimit',
+        chainId: generateWagmiChain(chainId),
       },
     ],
   });
 
-  const value = !!data
-    ? (parseInt(data.toString()) / 10 ** decimals).toFixed(6)
+  let pendingReward = 0;
```
Comment (MikeMurpher) [inline lib/hooks/usePrinterPoolData.tsx:35]: @cryptod00c check if user has wallet connected through wagmi

**Comment by cryptod00c** [comment]
@MikeMurpher did you figure this out?

**Comment by MikeMurpher** [comment]
@cryptod00c Yep, but closed it to keep the tiles instead of table

**Pull request closed without merging**